### PR TITLE
Update to `regex-automata@0.4.7`

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: provider/data/list/fingerprints.csv [hunk 1]
```diff
@@ -1,380 +1,380 @@
...
+list/and@2, da, 24B, 32914f71c9d587ff
+list/and@2, de, 25B, a0118012fae3e0a1
+list/and@2, doi, 40B, 1b95db885132a257
+list/and@2, dsb, 23B, 3bb37689fa2614f8
+list/and@2, el, 28B, 35c2f2f0277a33c8
+list/and@2, en, 47B, d5c0db9263bf647e
+list/and@2, en-001, 25B, b8523b7cd574b5b
+list/and@2, en-CA, 25B, -> en-001
+list/and@2, en-IN, 34B, 61f7e3861df4623e
+list/and@2, es, 1051B, bb8c9fc67206f5fd
```
Comment (robertbastian) [inline provider/data/list/fingerprints.csv:30]: not a fan of this +50%

File: provider/bikeshed/data/debug/list/or@2/es-AR.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
       "index_1": 3
     },
     "special_case": {
-      "condition": "o|ho|8|(11([\\.  ]?[0-9]{3})*(,[0-9]*)?([^\\.,[0-9]]|$))",
+      "condition": "^[oO]|[hH][oO]|8|(?:11(?:[\\.  ]?[0-9]{3})*(?:,[0-9]*)?(?:[^\\.,[0-9]]|$))",
```
Comment (sffc) [inline provider/bikeshed/data/debug/list/or@2/es-AR.json:18]: Note: The regex changed. Might account for the size diff. Something about case sensitivity?

File: provider/bikeshed/src/list/mod.rs [hunk 3]
```diff
@@ -53,26 +53,27 @@ fn load<M: DataMarker<Yokeable = ListFormatterPatternsV1<'static>>>(
...
     if langid.language == language!("es") {
-        if M::INFO == AndListV1Marker::INFO || M::INFO == UnitListV1Marker::INFO {
+        if M::INFO == AndListV2Marker::INFO || M::INFO == UnitListV2Marker::INFO {
             static I_SOUND: OnceLock<SerdeDFA<'static>> = OnceLock::new();
 
             // Starts with i or (hi but not hia/hie)
             let i_sound = I_SOUND.get_or_init(|| {
-                SerdeDFA::new(Cow::Borrowed("i|hi([^ae]|$)")).expect("Valid regex")
+                SerdeDFA::new(Cow::Borrowed("^[iI]|(?:[hH][iI](?:[^aeAE]|$))"))
+                    .expect("Valid regex")
```
Comment (sffc) [inline provider/bikeshed/src/list/mod.rs:62]: Issue: Why are you changing the regex in the same PR? If you want to change it, seems like that should be a separate PR we can discuss in more detail. It seems like this problem could be solved via other means, for example.

File: components/list/src/provider/serde_dfa.rs [hunk 4]
```diff
@@ -128,23 +128,33 @@ impl<'data> SerdeDFA<'data> {
     /// Creates a `SerdeDFA` from a regex.
```
Comment (sffc) [inline components/list/src/provider/serde_dfa.rs:128]: Thought: `DFA::from_bytes` didn't change from 0.2 to 0.4 ?

**Review by sffc — COMMENTED** [review]
Did you try 0.3?

File: provider/bikeshed/src/list/mod.rs (hunk 3 above)
Comment (robertbastian) [reply provider/bikeshed/src/list/mod.rs:62]: 0.4 requires an extra feature to support case-insensitive, and I was hoping I could get some size regression back like this. It's done now, I'd rather not redo all of this.

File: components/list/src/provider/serde_dfa.rs (hunk 4 above)
Comment (robertbastian) [reply components/list/src/provider/serde_dfa.rs:128]: define change?

File: provider/bikeshed/data/debug/list/or@2/es-AR.json (hunk 2 above)
Comment (robertbastian) [reply provider/bikeshed/data/debug/list/or@2/es-AR.json:18]: I changed the regex

File: components/list/src/provider/serde_dfa.rs (hunk 4 above)
Comment (sffc) [reply components/list/src/provider/serde_dfa.rs:128]: I was just expecting to see changes in how we deserialize the bytes but I only saw changes in how we build them.
